### PR TITLE
dev-libs/openssl: bump to 1.0.2h

### PR DESCRIPTION
**Pull request by crawford**

### CODE REVIEW

File: dev-libs/openssl/openssl-1.0.2h.ebuild [hunk 1]
```diff
@@ -12,23 +12,16 @@ HOMEPAGE="http://www.openssl.org/"
 SRC_URI="mirror://openssl/source/${MY_P}.tar.gz"
 
 LICENSE="openssl"
+# subslot set to 1.0.2g version as this is the first release without SSLv2
+# support and thus breaks nearly every openssl consumer (see bug #575548)
```
Comment (marineam) [inline dev-libs/openssl/openssl-1.0.2h.ebuild]: Sublot comment not applicable

File: dev-libs/openssl/openssl-1.0.2h.ebuild [hunk 2]
```diff
@@ -241,3 +232,17 @@ multilib_src_install_all() {
...
 	keepdir ${SSL_CNF_DIR}/private
 }
+
+pkg_preinst() {
+	has_version ${CATEGORY}/${PN}:0.9.8 && return 0
+	preserve_old_lib /usr/$(get_libdir)/lib{crypto,ssl}.so.0.9.8
+}
+
+pkg_postinst() {
+	ebegin "Running 'c_rehash ${EROOT%/}${SSL_CNF_DIR}/certs/' to rebuild hashes #333069"
```
Comment (marineam) [inline dev-libs/openssl/openssl-1.0.2h.ebuild]: Please leave out these two upgrade compatibility functions, I intentionally removed them

File: dev-libs/openssl/openssl-1.0.2h.ebuild (hunk 1 above)
Comment (crawford) [inline dev-libs/openssl/openssl-1.0.2h.ebuild]: Yeah, I was a little confused by that. They have that comment in upstream, but don't appear to specify a subslot.

**Comment by crawford** [comment]
Updated.

**Comment by marineam** [comment]
LGTM, I hope

**Pull request merged**